js/src/vm/PortableBaselineInterpret.cpp

```diff
@@ -4402,13 +4402,18 @@ static PBIResult PortableBaselineInterpret(JSContext* cx_, State& state,
 
     CASE(CheckReturn) {
       Value thisval = POP().asValue();
-      if (frame->returnValue().isObject()) {
-        PUSH(StackVal(frame->returnValue()));
-      } else if (!ret->isUndefined()) {
+      // inlined version of frame->checkReturn(thisval, result) (js/src/vm/Stack.cpp)
```

Comment:
I think it's fine to keep just this reference and remove the explanatory details below -- pushing exit frames for VM calls and jumping to the error-path on errors isn't otherwise unique to this case. (I should absolutely add more comments documenting these general idioms, though -- I'll take that as a TODO!)